src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -348,14 +353,34 @@ public Launcher createLauncher(TaskListener listener) {
                             ModelHyperlinkNote.encodeTo("/computer/" + getNodeName(), getNodeName()),
                             getTemplate().getDisplayName())
                     );
-                    listener.getLogger().println(getTemplate().getDescriptionForLogging());
+                    printAgentDescription(listener);
                     checkHomeAndWarnIfNeeded(listener);
                 }
             }
         }
         return launcher;
     }
 
+    @CheckForNull
+    public Pod getPod() {
+        return pod;
+    }
+
+    public boolean isShowRawYaml() {
+        return showRawYaml;
+    }
+
+    void assignPod(@CheckForNull Pod pod, boolean showRawYaml) {
+        this.pod = pod;
+        this.showRawYaml = showRawYaml;
+    }
+
+    private void printAgentDescription(TaskListener listener) {
+        if (pod != null && showRawYaml) {
```

Comment:
Since `pod` is only used when `showRamYaml` is set, you could simplify and just have the one `pod` field: `KubernetesLauncher` would simply decline to call `assignPod` unless `template.isShowRawYaml()`.